src/lib/hooks/useBlockNumber.tsx

```diff
@@ -62,7 +84,11 @@ export function BlockNumberProvider({ children }: { children: ReactNode }) {
         .catch((error) => {
           console.error(`Failed to get block number for chainId ${activeChainId}`, error)
         })
-
+      if (mainnetBlock === undefined) {
```

Comment:
I like that you are only updating this if it is undefined. Can you precede this line with a comment explaining why this is only necessary when it is `undefined`?